### ISSUE TITLE
(Content v2) <h1></h1> heading tags rendered as <h-1>

### ISSUE THREAD

**Issue by nwittwer** [nuxt/content#1222]
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.13.1`
- Nuxt Version:     `3.0.0-rc.3`
- Package Manager:  `yarn@1.22.17`
- Builder:          `vite`
- User Config:      `components`, `publicRuntimeConfig`, `head`, `loading`, `buildModules`, `content`, `image`, `modules`, `build`
- Runtime Modules:  `@nuxtjs/pwa@3.3.5`, `@nuxt/content@2.0.1`
- Build Modules:    `@nuxtjs/color-mode@3.1.3`, `@nuxtjs/eslint-module@3.1.0`, `@nuxtjs/tailwindcss@5.1.2`

### Reproduction

[https://codesandbox.io/s/nuxt3-contentv2-reproduction-w2j88p?file=/content/index.md](https://codesandbox.io/s/nuxt3-contentv2-reproduction-w2j88p?file=/content/index.md)

### Describe the bug

HTML `h1`, `h2`, etc. tags are converted into tags like `<h-1>`, and nest unexpected following lines instead of being closed immediately.

This was working correctly in Nuxt Content v1.

<img width="1122" alt="Screen Shot 2022-06-06 at 6 57 47 PM" src="https://user-images.githubusercontent.com/6360459/172208890-32f11f82-a1d8-40f3-8b9c-8b9c3f8d9d81.png">

### Additional context

Using HTML tags _inside_ of Markdown files can be useful for scenarios like using TailwindCSS classes (e.g. `.text-2xl`) to style headings or specific text blocks with custom styles or other HTML attributes.

More info: https://stackoverflow.com/questions/1058933/can-i-define-a-class-name-on-paragraph-using-markdown

### Logs

```
[Vue warn]: Failed to resolve component: H1
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
```

**Comment by king-11** [comment]
is it planned to be included in next release 😅

**Comment by nwittwer** [comment]
@king-11 Released in v2.1.0: https://github.com/nuxt/content/releases/tag/v2.1.0
